woocommerce/woocommerce-ios
# Login: New Jetpack required screen

Part of #8075


### Description

This PR adds a few UI changes to prepare for the native Jetpack setup flow. The design can be found here: ERGtCFLyiC9CFpHsw1XIid-fi-1110%3A37319. 

Changes include:
- New temporary feature flag for the native Jetpack setup flow. This should be replaced with an A/B test later.
- Handle navigation to the native Jetpack setup flow when the user enters a site address in the site discovery flow (from the store picker) and the site doesn't have Jetpack or is not connected to the user's WP.com account.
- Add a new screen for the Jetpack error reusable for both Jetpack installation and connection.
- Also: update the empty store picker screen following the new design.

### Testing instructions

Prerequisite: Make sure that you have a test store that is not associated with your WP.com account or doesn't have Jetpack.

- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- Notice that you are navigated to a new Jetpack required screen matching the Figma design. Ensure that the copy and CTA is correct based on the state of Jetpack on your site.


### Screenshots

| Jetpack not installed or activated | Jetpack connection missing |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/202086738-74a3e2e4-e576-4dd4-8e6d-281db40281bd.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/202086827-e169848d-1aa3-43b0-a1ea-8182e2071618.png" width=320 /> |

Empty store picker:
| Before | After |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/202121417-7d258d31-7213-4058-ac73-aef48e149e69.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/202120720-d7f8b3cc-2925-4cac-b1dd-32684404bd20.png" width=320 /> |

---
